# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: gfx/cairo/cairo/src/cairo-toy-font-face.c
# Commit: 3cdfb39376b7
# Full Hash: 3cdfb39376b736597fe658a322389117686b7e82
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2024-05-07 09:13:07
# Regressor Bug: 1892913
# File Overlap Count: 3
# Description:
#   Bug 1892913 - patch 1 - Update cairo source to 1.18.0 r=gfx-reviewers,lsalzman
#   
#   Wholesale replacement of files in gfx/cairo/cairo with contents of the 1.18.0 release tarball,
#   omitting subdirectories that are not part of the core library build.
#   
# ==============================================================================

diff -r 9daa351f875b -r 3cdfb39376b7 gfx/cairo/cairo/src/cairo-toy-font-face.c
--- a/gfx/cairo/cairo/src/cairo-toy-font-face.c	Mon May 06 19:22:35 2024 +0000
+++ b/gfx/cairo/cairo/src/cairo-toy-font-face.c	Mon May 06 19:25:12 2024 +0000
@@ -135,7 +135,7 @@
 			       cairo_font_slant_t     slant,
 			       cairo_font_weight_t    weight)
 {
-    unsigned long hash;
+    uintptr_t hash;
 
     key->family = family;
     key->owns_family = FALSE;
@@ -145,8 +145,8 @@
 
     /* 1607 and 1451 are just a couple of arbitrary primes. */
     hash = _cairo_hash_string (family);
-    hash += ((unsigned long) slant) * 1607;
-    hash += ((unsigned long) weight) * 1451;
+    hash += ((uintptr_t) slant) * 1607;
+    hash += ((uintptr_t) weight) * 1451;
 
     key->base.hash_entry.hash = hash;
 }
@@ -235,7 +235,7 @@
  * @weight: the weight for the font
  *
  * Creates a font face from a triplet of family, slant, and weight.
- * These font faces are used in implementation of the the #cairo_t "toy"
+ * These font faces are used in implementation of the #cairo_t "toy"
  * font API.
  *
  * If @family is the zero-length string "", the platform-specific default
@@ -340,7 +340,6 @@
  UNWIND:
     return (cairo_font_face_t*) &_cairo_font_face_nil;
 }
-slim_hidden_def (cairo_toy_font_face_create);
 
 static cairo_bool_t
 _cairo_toy_font_face_destroy (void *abstract_face)
@@ -469,7 +468,6 @@
     }
     return toy_font_face->slant;
 }
-slim_hidden_def (cairo_toy_font_face_get_slant);
 
 /**
  * cairo_toy_font_face_get_weight:
@@ -496,7 +494,6 @@
     }
     return toy_font_face->weight;
 }
-slim_hidden_def (cairo_toy_font_face_get_weight);
 
 static const cairo_font_face_backend_t _cairo_toy_font_face_backend = {
     CAIRO_FONT_TYPE_TOY,